test/hotspot/jtreg/TEST.groups

```diff
@@ -71,6 +71,9 @@ hotspot_native_sanity = \
 hotspot_containers = \
   containers
 
+hotspot_ir_transformations = \
+  ir_transformations
+
```

Comment:
as these tests are c2 tests, they are to be inside `test/hotspot/jtreg/compiler/c2`. I don't think `ir_framework` is a good directory name, as they aren't tests for `ir_framework`, nor `ir_framework` itself,  so I'd suggest `test/hotspot/jtreg/compiler/c2/ir_transformations`, or `test/hotspot/jtreg/compiler/c2/idealization`.

on a separate note, @JohnTortugo why have you added a dedicated group for these tests?

-- Igor

Reply:
Thanks Igor for your input!

> as they aren't tests for `ir_framework`

Right. This might be misleading. 

>  so I'd suggest test/hotspot/jtreg/compiler/c2/ir_transformations, or test/hotspot/jtreg/compiler/c2/idealization.

Could we also choose `test/hotspot/jtreg/compiler/c2/ir_tests` (should we use camel case or underscore?) as top folder for any kinds of IR tests? Then we could still add a subfolder like `ideal_transformations` to it (`test/hotspot/jtreg/compiler/c2/ir_tests/ideal_transformations`). If there are other IR tests later, for example for string opts, we can add a `stringopts` subfolder into the `ir_tests` top folder and so on.

Reply:
> Could we also choose `test/hotspot/jtreg/compiler/c2/ir_tests` as top folder for any kinds of IR tests? 

sounds good to me. 

> should we use camel case or underscore?

given it's java code, and the directory names are part of a package name, we should use lowerCamelCase.

Reply:
I am **against** moving tests under `/c2` directory. This directory was created to split original set of tests to run in parallel and not related to C2 compiler. I want to have directories in `/compiler` per optimization/transformation type as we have now. For example, `/loopopts` is for C2 only but it is not under `/c2`.

Actually I want to clear `c2` and `c1` directories eventually to move tests to corresponding category directories instead.

So I am for `/compiler/irTests`.

Reply:
> This directory was created to split original set of tests to run in parallel and not related to C2 compiler

aren't you mixing this up w/ `tier1_compiler_[1-3]` test group? if you aren't, this was the wrong way to split tests and we need to fix that.

>  I want to have directories in `/compiler` per `optimization/transformation` type as we have now. For example, `/loopopts` is for C2 only but it is not under `/c2`.
> Actually I want to clear `c2` and `c1` directories eventually to move tests to corresponding category directories instead.

I would prefer to clear `/compiler` directory and move c1-specific tests into `/compiler/c1` directory and c2-specific tests into `/compiler/c2` with optimization/transformation type being subdirectories inside `/compiler/c[12]` (and shared/common tests moved to `/compiler/shared`), so `loopopts` tests are to moved to `/compiler/c2/loopopts`. one of the reasons, most of optimizations tests will be compiler-specific and having them together in `/compiler` doesn't make much sense and just makes it harder to navigate/use test base. e.g. if we have all c2 tests in `/compiler/c2`, one can easily run them together to test c2 changes, if c2-specific tests are in several directories in `/compiler`, so one will have to have intrinsic knowledge about all tests in `/compiler` to select the right one.

Reply:
Should I move all  tests created in this PR to folders under `/compiler/c2/`? Separated by type of optimization - loopOpts, scalarReplacement, etc?

Reply:
I moved the tests to `/compiler/c2/irTests/`. Please let me know if the split into subfolders I did is reasonable.